tests/Integration/PrestaShopBundle/Routing/Converter/RoutingCacheKeyGeneratorTest.php

```diff
@@ -41,32 +43,35 @@ class YamlRoutesInModuleTest extends KernelTestCase
     /**
      * {@inheritdoc}
      */
-    protected function setUp()
+    protected function setUp(): void
     {
-        $kernel = self::createKernel();
-        $kernel->boot();
-
-        self::$container = $kernel->getContainer();
+        self::bootKernel();
 
-        HelperModule::addModule('demo');
+        $dirResources = dirname(__DIR__, 4);
+        if (is_dir($dirResources . '/Resources/modules_tests/demo')) {
+            Tools::recurseCopy($dirResources . '/Resources/modules_tests/demo', _PS_MODULE_DIR_ . '/demo');
+        }
 
-        $this->module = self::$container->get('prestashop.core.admin.module.repository')->getModule('demo');
+        $this->module = self::$kernel->getContainer()->get('prestashop.core.admin.module.repository')->getModule('demo');
         $this->module->onInstall();
-        self::$container->get('prestashop.core.cache.clearer.cache_clearer_chain')->clear();
+        self::$kernel->getContainer()->get('prestashop.core.cache.clearer.cache_clearer_chain')->clear();
     }
 
     /**
      * {@inheritdoc}
      */
-    protected function tearDown()
+    protected function tearDown(): void
     {
-        HelperModule::removeModule('demo');
+        $dirResources = dirname(__DIR__, 4);
```

Comment:
don't we need to call parent::tearDown to shutdown kernel ?

Reply:
Nope we add & remove module at the fly.